crates/router/src/connector/bitpay/transformers.rs

```diff
@@ -128,7 +128,7 @@ impl<F, T>
             .data
             .url
             .map(|x| services::RedirectForm::from((x, services::Method::Get)));
-        let connector_id = types::ResponseId::ConnectorTransactionId(item.response.data.id);
+        let connector_id = types::ResponseId::ConnectorTransactionId(item.response.data.id.clone());
```

Comment:
Add a field `order_id` in `BitpayPaymentResponseData` and use it in connector_response_reference_id